ParabolInc/action
# guard against case 2 drag from underneath case 3 1 user 2 tabs

TEST

CASE 2
- start a retro, add 3+ reflections, go to group phase
- Browser 1 drops card A on card B
- [x] Browser 2 picks up card B before card A completes its drop on it  (you have 1 second, go fast!) and drops it on card C
- the board is as expected: card A is in a group by itself, card B & C are in a group
Note: I added a 1 second delay on the server to make testing this easier. We'll need to remove that before we merge this!

CASE 3
- open 2 tabs for the same user
- in tab 1, drop card A on card B
- quickly try to pick up card A & ungroup it. keep trying until you have success
- [x] tab 2 does not get an "interception" toast
- [ ] tab 2 can drag card A  

AutoGroup
- ungroup all the cards
- click autogroup
- [x] the groups autogroup themselves & a refresh is not needed

## Notes

OMG. This is one million times better. Incredible work!

I pounded on this hard this morning. I still see a case every once in a great while where a frame must drop and one of the clients gets a hanging card. It is fixed on a refresh. Of course, we know that's possible so nothing to do about it now.

A couple of other things of note:

### Cards not reflowing on window resize event?

See gif below:

Is this is a regression? Not that the cards do not reflow when I resize, and must horizontal scroll to access them:

[gif](http://g.recordit.co/ArdDVSA2OE.gif)

^ looks like I can no longer embed big gifs :(

Shall we fix this now?

### Auto-grouping one-time failure

I ran 3 retros, each with different card content. On one run, I hit autogroup and about half of my cards disappeared. A refresh did not fix it. Since then, I've been unable to reproduce it.

Very odd. Something to keep an eye on. If we make a subsequent revision of this PR, I'll try to reproduce it.
